File: #209_Minimum_Size_Subarray_Sum/Solution.java
public class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int i = 0; int j = 0;
        int result = nums.length + 1;
        int sum = nums[j];
        while (true) {
        	if (i == j && sum >= s) return 1;
        	if (sum < s) {
        		if (j == nums.length - 1) break;
        		sum += nums[++j];
        	}
        	else {
        		result = Math.min(j - i + 1, result);
        		sum -= nums[i++];
        	}
        }
        return (result == nums.length + 1) ? 0 : result;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[] t = {2,3,1,2,4,3};
    	System.out.println(sol.minSubArrayLen(7, t));
    }
}

File: #209_Minimum_Size_Subarray_Sum/Solution.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
        	return 0
        i, j = 0, 0
        sum = 0
        minLen = len(nums) + 1
        while True:
        	if sum < s:
        		if j >= len(nums):
        			break
        		sum += nums[j]
        		j += 1
        	if sum >= s:
        		minLen = min(minLen, j - i)
        		sum -= nums[i]
        		i += 1
        if minLen == len(nums) + 1:
        	return 0
        return minLen

sol = Solution()
print sol.minSubArrayLen(7, [2,3,1,2,4,3])